### PR TITLE
Check for provisioning state during location polling

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
There are a few corner-cases for location polling where ARM can reply
with a provisioning state, so use it if available.
Fixed a test bug for body polling.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/azcore/internal/pollers/loc/loc.go [hunk 1]
```diff
@@ -84,15 +83,22 @@ func (p *Poller[T]) Poll(ctx context.Context) (*http.Response, error) {
...
-			return pollers.StatusInProgress, nil
-		} else if p.CurState > 199 && p.CurState < 300 {
+		if provState != "" {
+			p.CurState = provState
+		} else if resp.StatusCode == http.StatusAccepted {
+			p.CurState = pollers.StatusInProgress
+		} else if resp.StatusCode > 199 && resp.StatusCode < 300 {
 			// any 2xx other than a 202 indicates success
-			return pollers.StatusSucceeded, nil
+			p.CurState = pollers.StatusSucceeded
```
Comment (chlowell) [inline sdk/azcore/internal/pollers/loc/loc.go:97]: Is the comment correct? Do we somehow not land here when `resp.StatusCode == 202`?

**Review by chlowell — APPROVED** [review]
